src-java/kilda-persistence-neo4j/src/test/java/org/openkilda/persistence/repositories/impl/Neo4jFlowEventRepositoryTest.java

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.openkilda.model.history.FlowEvent;
+import org.openkilda.persistence.Neo4jBasedTest;
+import org.openkilda.persistence.repositories.history.FlowEventRepository;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Neo4jFlowEventRepositoryTest extends Neo4jBasedTest {
+
+    private static final String FLOW_1 = "FLOW_1";
+    private static final String ACTION_1 = "ACTION_1";
+    private static final String ACTION_2 = "ACTION_2";
+    private static final String ACTION_3 = "ACTION_3";
+    private static final Instant TIME_1 = Instant.parse("2020-07-06T11:04:41Z");
+    private static final Instant TIME_2 = Instant.parse("2020-07-06T11:04:41.482Z");
+    private static final Instant TIME_3 = Instant.parse("2020-07-06T11:04:42.321Z");
+
+    private static FlowEventRepository repository;
+
+    @BeforeClass
+    public static void setUp() {
+        repository = new Neo4jFlowEventRepository(neo4jSessionFactory, txManager);
+    }
+
+    @Test
+    public void findByFlowIdAndTimeFrameOrderTest() {
```

Comment:
I'd add test for the case with 0 millisecond part of the timestamp.

Reply:
It's already there.  `TIME_1` is a timestamp with 0 milliseconds. On previous code version this test  fails.